### PR TITLE
allow replacing template parts via env

### PR DESCRIPTION
**Pull request by grosser**
for ShardedDeployment I need to configure `spec.shards.list` with an array, I could add custom logic for that, but that does not really scale well ... so how about something generic ?

/cc @zendesk/samson @zendesk/compute

### CODE REVIEW

**Comment by adammw** [comment]
seems a little strange.. what's the usecase/story?

**Comment by grosser** [comment]
see PR description ;)

File: plugins/kubernetes/README.md [hunk 1]
```diff
@@ -190,3 +190,13 @@ out of rotation (alternatively set `metadata.annoations.container-nameofcontaine
...
 This can be useful for Migrations (see above).
 (On failure, samson always shows all pod logs)
+
+### Changing templates via ENV
+
+For custom things that need to be different between environments/deploy-groups.
+
+Use an annotation to configure what will to be replaced:
+```
+metadata.annoations.samson/set_via_env_json/metadata.labels.custom: SOME_ENV_VAR
```
Comment (adammw) [inline plugins/kubernetes/README.md]: typo

**Comment by grosser** [comment]
needed to change to `samson/set_via_env_json-` since there can only be a single / in an annotation key ...

**Review by adammw — APPROVED** [review]

**Pull request merged**